usr/src/cmd/iscsid/iscsi-initiator

```diff
@@ -170,6 +171,23 @@ mount_iscsi() {
 }
 
 umount_iscsi () {
+	# In case any non-"zones" zpools reside on iscsi drives, export
+	# all of them now so that the system won't hang later in the shutdown
+	# sequence when those zpools need to be unmounted.
+	zpools=$(zpool list -Ho name)
+	if [ -n $zpools ]; then
```

Comment:
I think we can get away with not including this check. The 'for pool in $zpools' loop won't run if there weren't any pools returned. But it's not a big deal if we leave this in.